helper-scripts/migrate.py

```diff
@@ -9,28 +9,95 @@
 
 This script runs those migration scripts, as needed, based on the current and
 target version of 4CAT. It optionally also pulls the latest version of 4CAT
-from Github.
+from Github and restarts the backend and frontend.
 """
 import subprocess
+import requests
 import argparse
 import shutil
+import shlex
 import time
+import json
 import sys
 import os
 import re
 
 from pathlib import Path
 
 
+def get_versions(target_version_file, current_version_file):
+	"""
+	Get versions
+
+	:return tuple:  (target version readable, target version comparable, current version r, current version c)
+	"""
+	if not current_version_file.exists():
+		# this is the latest version lacking version files
+		current_version = "1.9"
+	else:
+		with current_version_file.open() as handle:
+			current_version = re.split(r"\s", handle.read())[0].strip()
+
+	if not target_version_file.exists():
+		print("No VERSION file available. Cannot determine what version to migrate to.\n")
+		exit(1)
+
+	with target_version_file.open() as handle:
+		target_version = re.split(r"\s", handle.read())[0].strip()
+
+	current_version_c = make_version_comparable(current_version)
+	target_version_c = make_version_comparable(target_version)
+
+	return (target_version, target_version_c, current_version, current_version_c)
+
+
 def make_version_comparable(version):
+	"""
+	Make a version comparable with normal operators
+
+	:param str version:
+	:return str:
+	"""
 	version = version.strip().split(".")
 	return version[0].zfill(3) + "." + version[1].zfill(3)
 
 
+def finish(args):
+	"""
+	Finish migration
+
+	We might want to finish without running everything in the migration, so
+	this is made a function that can be called from any point in the script to
+	wrap up and exit.
+	"""
+	print("\n- Migration finished.")
+	print("  You can now safely restart 4CAT.\n")
+
+	if args.restart:
+		print("- Triggering a WSGI reload by touching 4cat.wsgi...")
+		Path("webtool/4cat.wsgi").touch()
```

Comment:
There is no 4cat.wsgi in the Docker setup. Which begs the question of how to restart the frontend in Docker.

Gunicorn runs with a `--reload` parameter and so if a file is updated the gunicorn workers are restarted. All I have to do is make a change to a file and refresh my browser when I make code changes to the frontend. (Note: I do not know if that should actually be the _default_ run setting.)

Reply:
I don't suppose this fail cause a failure though. It'll just create a file that isn't used, right?

Reply:
Well! That is actually even more convenient. What files does it watch, though? Can it be made to watch a specific file or should we `touch webtool/__init__.py` for example?

Reply:
The file is always there in any case (it's included in the repository), so even if touching it is harmless it doesn't really matter.

Reply:
Well, I probably shouldn't have been using `--reload` in production just for resource usage. https://docs.gunicorn.org/en/latest/settings.html?highlight=reload#reload
It must monitor all files that are loaded. I should probably change that (honestly I had forgotten while taking advantage of it frequently!). 

How to get it to reload via a command from the frontend (without reload enabled), I am not sure. The parameters are given to it via the `docker-compose.yml` file. We could restart Gunicorn _or_ the Docker container. That still leaves the question of how.